lib/ipaddr.rb

```diff
@@ -146,12 +194,20 @@ def ~
     return self.clone.set(addr_mask(~@addr))
   end
 
-  # Returns true if two ipaddrs are equal.
+  # Returns true if two ipaddrs are equal. Does not include the
+  # mask_addr in the check.
   def ==(other)
     other = coerce_other(other)
     return @family == other.family && @addr == other.to_i
   end
 
+  # Returns true if two ipaddrs are equal, including their
+  # mask_addrs
+  def equal_with_mask(other)
```

Comment:
> might be too long to be handy

Are there any guidelines on what is considered "too long"? The only possible shortening I see is in naming it `eql_with_mask?`, which makes it look more like `eql?`, which is not what this does. Dropping `with` from the name and I would interpret it as something that only compares the netmasks (so calling this with`10.0.0.1/24` and `192.168.0.1/24` would yield true, because the netmasks are the same. 

>  Also, it is not efficient to create two NetMask objects only to compare with each other

This sounds a bit like premature optimization to me. The alternatives I could think of are either making more methods public, using something like `other.instance_variable_get` or making the result something like `self == other && self.to_range == other.to_range`.
Option 1 would result in more public method, thus hiding less implementation details and making it more difficult to ever change the implementation. Option 2 feels kind of dirty. Option 3 does not exactly sound like a speedup.
A little speedup can be gained by caching the netmask (as suggested later in this review), but that doesn't exactly work if the other parameter is a String instead of an IPAddr, which I guess is the biggest speed killer. (Disclaimer: I didn't try to do any performance measurements)